packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -137,6 +137,7 @@ export class BlockList extends Component {
 					keyboardShouldPersistTaps="always"
 					scrollViewStyle={ {
 						flex: isRootList ? 1 : 0,
+						...( ! isRootList && { overflow: 'visible' } ),
```

Comment:
According to this [comment](https://github.com/WordPress/gutenberg/pull/20769#discussion_r397803057) I have checked and this line probably should be added to avoid borders crop. I also provided mentioned check to avoid broke bottom toolbar.

See screenshot below:
**without `visible: 'overflow'`**
<image src="https://user-images.githubusercontent.com/21242757/77577633-6df4b880-6ed7-11ea-8f15-dc1cb4067e48.png" width="400" />

**with applied `visible: 'overflow'`**
<image src="https://user-images.githubusercontent.com/21242757/77577715-8a90f080-6ed7-11ea-885a-4ac52494bf89.png" width="300" />

<image src="https://user-images.githubusercontent.com/21242757/77577769-a98f8280-6ed7-11ea-9dd7-5321b0e67897.png" width="400" />

Reply:
However Android need some more adjustment on that because it crop borders in Column, will check tomorrow what happens

<image src="https://user-images.githubusercontent.com/21242757/77587011-663d1000-6ee7-11ea-9eb1-7b7707b47e84.png" width="300" />

Reply:
It is fixed - some of the code from initial approach stays when merging new bordering logic and that was the cause of cropped borders on Android, thanks @dratwas for help !